src/paymentsLog.js

```diff
@@ -17,11 +21,79 @@ module.exports = (pgdb) => {
   server.post('/payments/paypal',
     bodyParser.urlencoded({extended: true}),
     async (req, res) => {
+      const { body } = req
       await pgdb.public.paymentsLog.insert({
         method: 'PAYPAL',
-        pspPayload: req.body
+        pspPayload: body
       })
-      return res.sendStatus(200)
+
+      // end connection
+      res.sendStatus(200)
+
+      // payPledge in case that didn't happen already:
+      // in case users close the paypal tab before being redirected back to us,
+      // we only get notified about the payment via this webhook.
+      // we ignore non Completed status changes for now
+      if (body.payment_status === 'Completed') {
+        const pledgeId = body.item_name
+
+        const transaction = await pgdb.transactionBegin()
+        let userId
+        try {
+          // load pledge
+          // FOR UPDATE to wait on other transactions
+          const pledge = (await transaction.query(`
+            SELECT *
+            FROM pledges
+            WHERE id = :pledgeId
+            FOR UPDATE
+          `, {
+            pledgeId
+          }))[0]
+
+          if (pledge && pledge.status !== 'SUCCESSFUL') {
+            userId = pledge.userId
+
+            const pspPayload = {
+              tx: body.txn_id, // normalize with redirect params
+              webhook: true,
+              ...body
```

Comment:
mv `tx: body.txn_id` and `webhook: true` after the spread if you want to ensure it's not overwritten